### ISSUE TITLE
[Discussion] Azure Functions Runtime 2.0.12095 release

### ISSUE THREAD

**Issue by paulbatum** [Azure/app-service-announcements-discussions#67]
Discussion issue for the Azure Functions runtime release 2.0.12095-alpha

**Comment by rasmustherkelsen** [comment]
I have logic apps that use functions pinned to 2.0.11961-alpha. Just as when you deployed 12050 I am unable to deploy those logic apps using ARM templates as they are unable to resolve the invoke URLs. The actual error code is is "FunctionInvokeUrlNotFound". Last time, it was almost for a week from you announced you started deploying until it worked again. Are you aware if this? It would be critical to my customer if this happened after you go GA. I also worked with Azure Support on this and they never came up with a reason for this failing.

**Comment by paulbatum** [comment]
We resolved one issue (https://github.com/Azure/azure-functions-host/issues/3303) that was impacting retrieval of the invoke URL in build 12080. However this version is not deployed globally yet. Did you verify whether your function app has upgraded to 12080?

**Comment by rasmustherkelsen** [comment]
Sounds good and looking forward to the release. My functions app is still on 12050 so awaiting rollout (running in North Europe datacenter). Thanks.

**Comment by paulbatum** [comment]
I have updated this issue to reflect the fact that we are proceeding with version 2.0.12093 (instead of 2.0.12080).

**Comment by paulbatum** [comment]
Its not yet. In fact the version we deploy globally might be slightly different, I'll be updating this post again in 12 hours or so. If you want to be on the latest just set your version to ~2 right now.

**Comment by odd-rune** [comment]
Does this affect how long 2.0.11961 is available? We have some issues with 2.0.12050 and would rater wait for 2.0.12093.

**Comment by paulbatum** [comment]
No, 2.0.11961 will still be available through till the end of September.

**Comment by igorgnedish** [comment]
Hello, after update to 2.0.12095 it seems that proxies have been broken. They return a 404 response even if the route is matched. Temporarily I have rolled back to 2.0.12050 and it started to work again. Please, ask me if you need more details regarding this.

PS: we have just a simple routing functions, without request/response overrides. In our case function app does not contain anything except proxies.json file. Tried to add a fake function - the same result.

**Comment by davidebbo** [comment]
@igorgnedish Thanks for reporting. Seems the issue only happens when the proxies.json is first created, but then everything works if you restart the app. Does that match your observation? We will investigate further.

**Comment by davidebbo** [comment]
@igorgnedish tracked by https://github.com/Azure/azure-functions-host/issues/3466.

**Comment by igorgnedish** [comment]
@davidebbo yes, it lookslike. Only one addition. Noticed if I restart function via start & stop -> proxies won't work. It the function is restarted via 'restart' -> proxies works.

**Comment by igorgnedish** [comment]
@davidebbo and also noticed that restart helps me not in all cases. Sometimes I have to repeat stop-start-restart steps a couple of times in order to get proxies working.

**Comment by odd-rune** [comment]
Slow triggering of queue messages. 
After upgrading to 2.0.12050, and later with 2.0.12095 we see that messages are on the queues for a longer time that expected before being processed, sometimes minutes. 
This is with the same setting in host.json, just moved to the new extensions section.
      "maxPollingInterval": 2000, "visibilityTimeout": "00:02:00", "batchSize": 1, "maxDequeueCount": 5 "newBatchThreshold": 1

**Comment by davidebbo** [comment]
@igorgnedish best to take the discussion to https://github.com/Azure/azure-functions-host/issues/3466. The fix will be deployed this week.

**Comment by paulbatum** [comment]
@rtorri01 Looks like you are hitting the breaking changes that were announced in https://github.com/Azure/app-service-announcements/issues/129.  If you run the `func extensions install` command locally (for more info see [here](https://docs.microsoft.com/en-us/azure/azure-functions/functions-triggers-bindings#local-development-azure-functions-core-tools)), you should see that it updates your extensions.csproj file to include a reference to `Microsoft.Azure.WebJobs.Extensions.Storage`.  Deploy this change and that should resolve that particular error. You might hit other issues depending on whether you are impacted by any of the other changes in that announcement.

**Comment by romanovs** [comment]
@paulbatum thanks for your answer! It helped me too. I'm wondering if we have to do all the steps from the instruction everytime for every new Function App?

**Comment by paulbatum** [comment]
@romanovs The instructions we shared are really about upgrading from earlier previews. If you're starting completely from scratch and you have local development tools that are in sync with what we've released on Azure, then things should just work without any special steps.

**Comment by rtorri01** [comment]
@paulbatum after following your recommendations and links to update, I can't see much information for the Azure function in **Java code** because most of them are for C# in Visual Studio, anyway, I tried to make all the changes but nothing, the above error is still happening.
Are you still working for new changes on Java code ?
Can you provide a pom.xml file as a template reference ?

Below, are the Application Settings that I am using:
![applicationsettings](https://user-images.githubusercontent.com/42978791/45703785-4afb8280-bb43-11e8-9312-6e60bc0e167a.png)

**Comment by paulbatum** [comment]
@rtorri01 please be careful about sharing screenshots of your app settings. You leaked part of your keys.
Your app settings are not the problem. As I mentioned before, your extensions.csproj file needs to be updated.

**Comment by rtorri01** [comment]
> @rtorri01 please be careful about sharing screenshots of your app settings. You leaked part of your keys.
> Your app settings are not the problem. As I mentioned before, your extensions.csproj file needs to be updated.

@paulbatum thanks for your recommendations. I don't have any extensions.csproj file deployed on Function app on "d:\home\site\wwwroot" directory, instead of that, I am using a JAR file because I have a Java project deployed.
Is there some way to update the packages for Java project ?
Thanks in advance.

**Comment by romanovs** [comment]
@rtorri01 , we use Java too. You can either create this file locally by 'func extensions install --package Microsoft.Azure.WebJobs.Extensions.Storage --version 3.0.0-beta8' or create right in Kudu Console and copy-paste what they provided in the instruction. Make sure '.Extensions.Storage' has version '3.0.0-beta8' and '.Script.ExtensionsMetadataGenerator' has version '1.0.0'. In 1st case you need to deploy your function with that file, then run 'dotnet build extensions.csproj -o bin --no-incremental --packages D:\home.nuget' and restart your Func App

**Comment by jmt4b04d4v** [comment]
@paulbatum , @romanovs thanks a lot for the tips and advice.

We were using and old version of the `'azure-functions-core-tools'`. We required to [update Node.js](https://tecadmin.net/upgrade-nodejs-via-npm/), then upgrade `'azure-functions-core-tools'` with `npm i -g azure-functions-core-tools@core --unsafe-perm true` and after that we were able to properly run `func extensions install --package Microsoft.Azure.WebJobs.Extensions.Storage --version 3.0.0-beta8` to add the extension to our existing project in our local development environments and deploy it to Azure. Using this approach we weren't required to build the referenced extensions in the Kudu console, all the required artifacts were generated with `mvn clean package`.

JIC, the `Microsoft.Azure.WebJobs.Extensions.Storage` extension covers [Blob (trigger/input/output)](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-blob), [Queue (trigger/output)](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-queue) and [Table (input/output)](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-table) bindings.

CC @rtorri01
